ui/src/components/comment-node.tsx

```diff
@@ -852,12 +884,20 @@ export class CommentNode extends Component<CommentNodeProps, CommentNodeState> {
     i.setState(i.state);
   }
 
-  handleDeleteClick(i: CommentNode) {
+  displayDeletionOrRestorationConfirmation(showConfirmation: boolean) {
+    return function (i: CommentNode) {
```

Comment:
Why is this returning a function here, just do this like all the show confirm dialogs. Don't have functions return functions.

Reply:
@dessalines  The idea was based on DRY. A lot of the functions do the same thing. By currying and passing in a Boolean flag, as I'm doing here, the code duplication is reduced. If that's not what the project wants, that's fine by me. I just thought it would help clean things up by introducing a new pattern.

Reply:
No meta-programming is necessary here. You need to look at the other examples in the code, in this very file.

```
  handleModBanShow(i: CommentNode) {
    i.state.showBanDialog = !i.state.showBanDialog;
    i.state.banType = BanType.Site;
    i.setState(i.state);
  }
```